decode.go

```diff
@@ -1381,8 +1379,7 @@ func (d *Decoder) decodeStruct(ctx context.Context, dst reflect.Value, src ast.N
 				if foundErr != nil {
 					continue
 				}
-				var te *errors.TypeError
-				if errors.As(err, &te) {
+				if te, ok := err.(*errors.TypeError); ok {
```

Comment:
I think this will stop working if the `err` is wrapped somewhere. Why did you change the sequence using As ?